src/wallet/spend.cpp

```diff
@@ -557,12 +557,15 @@ CallResult<SelectionResult> SelectCoins(const CWallet& wallet, const std::vector
             // Try with unlimited ancestors/descendants. The transaction will still need to meet
             // mempool ancestor/descendant policy to be accepted to mempool and broadcasted, but
             // OutputGroups use heuristics that may overestimate ancestor/descendant counts.
-            if (!fRejectLongChains) {
-                if (auto r8{AttemptSelection(wallet, value_to_select,
-                                      CoinEligibilityFilter(0, 1, std::numeric_limits<uint64_t>::max(), std::numeric_limits<uint64_t>::max(), true /* include_partial_groups */),
-                                      vCoins, coin_selection_params)}) {
-                    return r8;
+            if (auto r8{AttemptSelection(wallet, value_to_select,
+                                         CoinEligibilityFilter(0, 1, std::numeric_limits<uint64_t>::max(), std::numeric_limits<uint64_t>::max(), true /* include_partial_groups */),
+                                         vCoins, coin_selection_params)}) {
+                if (fRejectLongChains) {
```

Comment:
In c8a43eca7c4081f8ca87be1befbbcdbfdc798a10 "send: return proper error message for too-long-mempool-chain failure."

I'm not a huge fan of doing all of the coin selection computation just to immediately throw away the result. It seems like there should be a better way to do this, e.g. by changing how we fill `vCoins` to be guaranteed to have a solution at this point iff long chain utxos are included.

Reply:
hmm yeah, nice point. Will cook something for it.